From 311cb57e0fd0b5be271c92f3be5ecfe9b132b66c Mon Sep 17 00:00:00 2001
From: Vlastimil Babka <vbabka@suse.cz>
Date: Tue, 10 Jul 2018 13:00:27 +0200
Subject: [PATCH] xen/x86/traps: add missing kernel CR3 switch in bad_iret path
Patch-mainline: Never, SUSE-Xen specific
References: bsc#1098658

In error_bad_iret, we have user CR3 already but are about to execute kernel
code again starting with fixup_bad_iret(). We need to switch to kernel CR3.

Signed-off-by: Vlastimil Babka <vbabka@suse.cz>
Automatically created from "patches.suse/x86-traps-add-missing-kernel-CR3-switch-in-bad_iret-.patch" by xen-port-patches.py

--- a/arch/x86/kernel/entry_64-xen.S
+++ b/arch/x86/kernel/entry_64-xen.S
@@ -1327,6 +1327,7 @@ bstep_iret:
 
 error_bad_iret:
 	SWAPGS
+	SWITCH_KERNEL_CR3
 	mov %rsp,%rdi
 	call fixup_bad_iret
 	mov %rax,%rsp
